### PR TITLE
Z->mumu simultanious fitting for the 2016 MSSM analysis

### PR DESCRIPTION
**Pull request by rcaspart**
This extends the setup for the 2016 MSSM datacards with the possibility to include the z->mumu control regions in the fit (expected names are zmm_nobtag and zmm_btag with a muon-momentum scale shape uncertainty CMS_htt_scale_m_13TeV). It can be enabled by setting the --fit_zmm flag (the default setup is unchanged).

Currently outstanding updates are to the b-tagging and JEC uncertainties. These are currently based on the uncertainties in the other channels and will be dedicatedly measured for the zmm categories.

### CODE REVIEW

File: HIG16037/src/HttSystematics_MSSMRun2.cc [hunk 1]
```diff
@@ -69,6 +69,8 @@ void AddMSSMRun2Systematics(CombineHarvester & cb, int control_region = 0) {
   // Electron and muon efficiencies
   // ------------------------------
   cb.cp().AddSyst(cb, "CMS_eff_m", "lnN", SystMap<channel, process>::init
+    ({"zmm"}, {"ZTT", "ZLL", "TT", "VV", "ZL"},  1.04)
```
Comment (ajgilbert) [inline HIG16037/src/HttSystematics_MSSMRun2.cc]: Don't need ZLL here, but no harm

File: HIG16037/src/HttSystematics_MSSMRun2.cc [hunk 2]
```diff
@@ -169,6 +171,16 @@ void AddMSSMRun2Systematics(CombineHarvester & cb, int control_region = 0) {
     ({"et"}, {9}, {"TTJ"},   0.99,   1.01)
     ({"et"}, {9}, {ggH},  0.98, 1.02) 
     ({"et"}, {9}, {bbH},  0.98, 1.02) 
+    ({"zmm"}, {8}, {"VV"},   1.01,   0.99) 
+    ({"zmm"}, {8}, {"TT"},   1.03,   0.97) 
+    ({"zmm"}, {8}, {"bbH"},  1.01,   0.99)
```
Comment (ajgilbert) [inline HIG16037/src/HttSystematics_MSSMRun2.cc]: drop bbH here

File: HIG16037/src/HttSystematics_MSSMRun2.cc [hunk 3]
```diff
@@ -200,6 +212,11 @@ void AddMSSMRun2Systematics(CombineHarvester & cb, int control_region = 0) {
     ({"et"}, {9}, {"ZL"},   0.99,   1.01)
     ({"et"}, {9}, {"ZJ"},   0.98,   1.00)
     ({"et"}, {9}, {"ZTT"},  0.97,   1.03)
+    ({"zmm"}, {9}, {"ZTT"},  0.98,   1.02)
+    ({"zmm"}, {9}, {"ZLL"},  0.90,   1.05)
```
Comment (ajgilbert) [inline HIG16037/src/HttSystematics_MSSMRun2.cc]: No ZLL here

**Comment by adewit** [comment]
Ok so I tested this a bit and have two comments/questions:
firstly - the default T2W command from mssm_protocol (making workspaces for all subdirs created) crashes because of the lack of signals in the zmm subdirs. Can of course just run the command for each subdirectory individually, but there's probably also some easier workaround. In any case this needs to be documented in the protocol so anyone can use it
second - does this actually work if we just want to fit one channel? I only see z->mumu in the cmb datacards, but maybe I'm missing something...

**Comment by rcaspart** [comment]
I fixed it now such that we do not create a dedicated folder for zmm (which as @adewit pointed out requires special options to tell text2workspace to build a workspace without actual signal contribution). In addition it now properly adds the corresponding zmm regions to the other datacards folders (i.e. both zmm cateogries for the full channels and only no b-tag/b-tag for the dedicated ones).

**Pull request merged**